sql/src/main/java/io/crate/planner/operators/WindowAgg.java

```diff
@@ -74,6 +74,9 @@
             LinkedHashMap<WindowDefinition, ArrayList<WindowFunction>> groupedFunctions = new LinkedHashMap<>();
             for (WindowFunction windowFunction : windowFunctions) {
                 WindowDefinition windowDefinition = windowFunction.windowDefinition();
+                if (windowFunction.filter() != null) {
+                    columnsUsedInFunctions.add(windowFunction.filter());
```

Comment:
Would using `...add(LogicalPlanner.extractColumns(windowFunction.filter())` also work? Would have the advantage that we only collect the inner columns once.

E.g. in the test case. `AVG(x) FILTER (WHERE x > 1)` the `x` would be collected only once.

Reply:
But we need the filter symbol(function, select symbol) in `toCollect`,  not the columns that it uses. It won't work if we extract only columns.

Reply:
Why not? If `filter` itself keeps being a function, then the `InputColumns` should create the right `Function/InputColumn` structure.